scripts/index.js

```diff
@@ -0,0 +1,133 @@
+    temperatureText.textContent = `${state.currentTemp}`;
+    setTemperatureColor(state.currentTemp);
+    setLandscapeImage(state.currentTemp);
+};
+
+const setSkyImage = (sky) => {
+    const skyImage = document.getElementById("skyImage");
+    if (sky == "Sunny") {
+        skyImage.src='./images/sunny-sky.jpg';
+    } else if (sky == "Cloudy") {
+        skyImage.src='./images/cloudy-sky.jpg';
+    } else if (sky == 'Rain') {
+        skyImage.src='./images/rainy-sky.jpg';
+    } else if (sky == 'Snow') {
+        skyImage.src='./images/snowy-sky.jpg';
+    } else if (sky == 'Storm') {
+        skyImage.src='./images/storm-sky.jpg';
+    }
+}
+
+const setSky = () => {
+    const skyInput = document.getElementById("skySelect").value;
+    setSkyImage(skyInput);
+    state.sky = skyInput;
+}
+
+const setCity = () => {
+    const cityInput = document.getElementById("cityInput").value;
+    const cityHeading = document.getElementById("cityHeading");
+    cityHeading.textContent = cityInput;
+}
+
+const resetCity = () => {
+    const cityInputField = document.getElementById("cityInput");
+    cityInputField.value = state.defaultCity;
+    setCity();
+}
+
+////////// register event handlers ///////////
+
+
+const increaseTempButton = document.getElementById("increaseTemp");
+const decreaseTempButton = document.getElementById("decreaseTemp");
+
+increaseTempButton.addEventListener("click", increaseTemperature);
+decreaseTempButton.addEventListener("click", decreaseTemperature);
+
+const skySelection = document.getElementById("skySelect");
+skySelection.addEventListener("change", setSky);
+
+const citySelection = document.getElementById("cityInput");
+citySelection.addEventListener("input", setCity);
+
+const cityResetButton = document.getElementById("resetCityButton");
+cityResetButton.addEventListener("click", resetCity);
+
+
+/**
+ * For some reason, my javascript doesn't run when i try to register event handlers once DOM content is loaded.
+ * Below is my original, non-working code. Once I took the content of "registerEventHandlers" out of a function
+ * and just pasted it (above), it works. I'd love any insight. 
+ */
+// const registerEventHandlers = () => {
+//     const increaseTempButton = document.getElementById("increaseTemp");
+//     const decreaseTempButton = document.getElementById("decreaseTemp");
+    
+//     increaseTempButton.addEventListener("click", increaseTemperature);
+//     decreaseTempButton.addEventListener("click", decreaseTemperature);
+    
+//     const skySelection = document.getElementById("skySelect");
+//     skySelection.addEventListener("change", setSky);
+    
+//     const citySelection = document.getElementById("cityInput");
+//     citySelection.addEventListener("input", setCity);
+    
+//     const cityResetButton = document.getElementById("resetCityButton");
+//     cityResetButton.addEventListener("click", resetCity);
+// };
+
+// document.addEventListener("DOMContentLoaded", registerEventHandlers);
```

Comment:
I think this should have been:

```suggestion
// window.addEventListener("DOMContentLoaded", registerEventHandlers);
```

Reply:
I just Googled "DOMContentLoaded" and got this:  https://developer.mozilla.org/en-US/docs/Web/API/Window/DOMContentLoaded_event